core-common/src/main/java/org/zalando/nakadi/repository/kafka/AvroDeserializerWithSequenceDecoder.java

```diff
@@ -0,0 +1,42 @@
+package org.zalando.nakadi.repository.kafka;
+
+import org.apache.avro.generic.GenericRecord;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.zalando.nakadi.domain.EnvelopeHolder;
+import org.zalando.nakadi.service.AvroSchema;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class AvroDeserializerWithSequenceDecoder {
+
+    private final SequenceDecoder metadataSequenceDecoder;
+    private final SequenceDecoder eventSequenceDecoder;
+
+    public AvroDeserializerWithSequenceDecoder(final AvroSchema schemas) {
+        this.metadataSequenceDecoder = new SequenceDecoder(schemas.getMetadataSchema());
+        this.eventSequenceDecoder = new SequenceDecoder(schemas.getNakadiAccessLogSchema());
+    }
+
+    public byte[] deserializeAvro(final EnvelopeHolder envelop) throws RuntimeException {
```

Comment:
```suggestion
    public byte[] deserializeAvro(final EnvelopeHolder envelop) {
```
It's not a checked exception, so no need to declare it.